include/boost/math/concepts/real_concept.hpp

```diff
@@ -60,34 +60,34 @@ class real_concept
 {
 public:
    // Constructors:
-   real_concept() : m_value(0){}
-   real_concept(char c) : m_value(c){}
+   real_concept() noexcept : m_value(0){}
```

Comment:
I believe that it will fix the CI failures. `std::atomic` requires a `noexcept` constructor so attempting to use `Real = real_concept` yields:

```
/usr/include/c++/9/atomic:198:7: note: ‘std::atomic<_Tp>::atomic() noexcept [with _Tp = boost::math::concepts::std_real_concept]’ is implicitly deleted because its exception-specification does not match the implicit exception-specification ‘’
```

If this throws off your conceptual requirements I think something to the effect of `std::enable_if_t<std::is_nothrow_constructible<Real>, bool> = true` would work. Either way I am now seeing that this has the potential to break user code.

Reply:
Sorry, just to say, real_concept shouldn't change, at least not in this way - it doesn't model the required concept any more.

Reply:
Yeah, recovering even double precision with a Monte-Carlo integration already takes roughly the age of the universe, so I don't see any particularly compelling reason to make sure it works with multiprecision.

Reply:
I went with option 1 and reduced the allowable types to those that are trivially copyable (the concept requirement for `std::atomic`). This is reflected in the docs as well.